src/storage/txn/tracker.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright 2025 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cell::RefCell;
+
+use tikv_util::time::Instant;
+use tracker::{FutureTrack, TrackerToken, GLOBAL_TRACKERS};
+
+use crate::storage::metrics::CommandKind;
+
+thread_local! {
+    // The TLS state exists solely for the purpose of [`TlsFutureTracker::collect_to_tracker`]
+    // Remove this complication if we can find a better way to collect the time.
+    static CURRENT_STATE: RefCell<Option<State>> = const { RefCell::new(None) };
+}
+
+#[derive(Debug, Clone, Copy)]
+enum PollState {
+    None,
+    Begin(Instant),
+    Finish(Instant),
```

Comment:
It is a littble bit confusing with `None` and `Finish` and the `PollState` is initialized with `Finish` but not `None`, how about setting state like this:
- `None` for new object
- `Begin` after first poll
- `IDLE` when stopped
- `Finish` when all done
and do not record duration after `Finish` state.

Reply:
Will add more comments.
The `PollState` is not about tracking the lifetime of a future object, but for the `Future::poll` method.